common/errors.go

```diff
@@ -5,16 +5,54 @@ import (
 	"runtime"
 )
 
+//----------------------------------------
+// Convenience methods
+
+// ErrorWrap will just call .TraceAt(), or create a new *cmnError.
+func ErrorWrap(cause interface{}, format string, a ...interface{}) Error {
+	msg := Fmt(format, a...)
+	if causeCmnError, ok := cause.(*cmnError); ok {
+		return causeCmnError.TraceAt(1, msg)
+	} else {
+		// NOTE: cause may be nil.
+		// NOTE: do not use causeCmnError here, not the same as nil.
+		return newError(msg, cause, cause).Stacktrace()
+	}
+}
+
 //----------------------------------------
 // Error & cmnError
 
+/*
```

Comment:
why not move this to `errors` subpackage? then it can be `errors.Wrap`

Reply:
tmlibs errors?  I'm down for that.

Reply:
well, because cmn is convenient...  if we want to have cmn, then it's more convenient to do cmn.ErrorWrap, than importing another module.

Reply:
I'd argue that Go2 should support Error natively... so I'd want it in cmn.

Reply:
Yeah.. `cmn.ErrorWrap` is fine too